src/transaction.c

```diff
@@ -334,7 +334,11 @@ int git_transaction_commit(git_transaction *tx)
 				return error;
 		}
 
-		if (node->ref_type != GIT_REFERENCE_INVALID) {
+		if (node->ref_type == GIT_REFERENCE_INVALID) {
+			/* ref was locked but not modified */
+			git_refdb_unlock(tx->db, node->payload, false, false, NULL, NULL, NULL);
```

Comment:
I think we want to check the return code here, don't we?

Reply:
@pks-t, thank you for the review. What kind of error handling do you propose? Just returning an error immediately on failed unlock would break committing the rest of the transaction, which is imho more important to avoid. Should we collect errors and then return it/them after the transaction? What about multiple errors? I'm a bit torn.

Reply:
In fact, I'd say to just return an error immediately. In essence, it really isn't that much different to the case where the reftype is valid:

1. We call  `update_target`
2. `update_target` returns an error immediately if `git_refdb_unlock` fails
3. `git_transaction_commit` aborts if `update_target` fails

So if you ask me, we should treat errors just the same in this new code and directly return the error code.

Reply:
Alright!

Reply:
Thanks! I guess this looks good to me now, but I'll wait for others to chime in. Please feel free to ping me again if you haven't received another answer in a few days